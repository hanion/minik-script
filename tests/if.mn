// if test

// then branch
a := 2;
if a == 2 {
	string := "then branch";
	print string;
} else if a == 1 {
	print "else if branch";
} else {
	print "else branch";
}


// else if branch
a = a - 1;
if a == 2 {
	string := "then branch";
	print string;
} else if a == 1 {
	print "else if branch";
} else {
	print "else branch";
}

// else branch
a = a - 1;
if a == 2 {
	string := "then branch";
	print string;
} else if a == 1 {
	print "else if branch";
} else {
	print "else branch";
}


// if scope test
b :: false;
if b {
	local_string := "local";
} else {
	print "should give undefined variable error ->";
	print local_string;
}


