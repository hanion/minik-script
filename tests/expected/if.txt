then branch
else if branch
else branch
should give undefined variable error ->
[ERROR] [line 44], Undefined variable 'local_string'.
